templatestore/views.py

```diff
@@ -424,13 +417,77 @@ def get_template_versions_view(request, name):
             status=404,
         )
 
+@csrf_exempt
+def get_tiny_url_from_db(request, name, version):
+    if request.method != "GET":
+        return HttpResponseBadRequest("invalid request method: " + request.method)
+    try:
+        templateTable=Template.objects.get(name=name)
+        try:
+            versionTable= TemplateVersion.objects.get(template_id_id=templateTable.id, version=version)
+            data = json.dumps(versionTable.tiny_url)
+            return HttpResponse(data)
+        except Exception:
+            raise (
+                Exception(
+                    "Validation: Template Version with version " + version + " does not exist"
+                )
+            )
+    except Exception:
+        raise (
+            Exception(
+                "Validation: Template with name " + name + " does not exist"
+            )
+        )
+    
+@csrf_exempt
+def save_tiny_url(request):
+    if request.method != "PUT":
+        return HttpResponseBadRequest("invalid request method: " + request.method)
+    # Get the required template
+    try:
+        data=json.loads(request.body)
+        try:
+            version=data['templateVersion']
+            templateTable=Template.objects.get(name=data['templateName'])
+            #get template version table
+            try:
+                versionTable= TemplateVersion.objects.get(template_id_id=templateTable.id,version=version)
+                versionTable.tiny_url=data['tinyUrlArray']
+                versionTable.save()
+            except Exception as e:
+                logger.exception(e)
+                return HttpResponse(
+                    json.dumps({"message": str(e)}),
+                    content_type="application/json",
+                    status=200,
+                )
+        except Exception as e:
+            logger.exception(e)
+            return HttpResponse(
+                json.dumps({"message": "Template doesnot exists"}),
+                content_type="application/json",
+                status=200,
+            )
+        return HttpResponse(
+            json.dumps({"message": "Saved successfully"}),
+            content_type="application/json",
+            status=200
+        )
+    
+    except Exception as e:
+        logger.exception(e)
+        return HttpResponse(
+            json.dumps({"message": str(e)}),
```

Comment:
don't throw the system error here. System error is used to debug in logs and acts as a security vulnerability. Throw a user understandable error here.